benchmark/core/Cabana_CommPerformance.cpp

```diff
@@ -348,20 +371,108 @@ void performanceTest( std::ostream& stream, const std::size_t num_particle,
                 comm_memory_space(), particles );
 
             s0 = Cabana::slice<0>( comm_particles );
-            Cabana::scatter( halo_fast, s0 );
+            Cabana::scatter( halo, s0 );
 
             s1 = Cabana::slice<1>( comm_particles );
-            Cabana::scatter( halo_fast, s1 );
+            Cabana::scatter( halo, s1 );
 
             s2 = Cabana::slice<2>( comm_particles );
-            Cabana::scatter( halo_fast, s2 );
+            Cabana::scatter( halo, s2 );
 
             s3 = Cabana::slice<3>( comm_particles );
-            Cabana::scatter( halo_fast, s3 );
+            Cabana::scatter( halo, s3 );
 
-            Cabana::deep_copy( particles, comm_particles );
+            particles = Cabana::create_mirror_view_and_copy(
+                data_memory_space(), comm_particles );
             halo_slice_scatter.stop( fraction );
         }
+
+        // Preallocated buffers.
+        // ---
+        double overallocation = 1.1;
+
+        // Run preallocated AoSoA buffer tests and time the ensemble.
+        auto comm_particles = Cabana::create_mirror_view_and_copy(
+            comm_memory_space(), particles );
+        auto gather = createGather( halo, comm_particles, overallocation );
+        for ( int t = 0; t < num_run; ++t )
+        {
+            halo_buffer_aosoa_gather.start( fraction );
+            auto comm_particles = Cabana::create_mirror_view_and_copy(
+                comm_memory_space(), particles );
+            gather.apply( comm_particles );
+            particles = Cabana::create_mirror_view_and_copy(
+                data_memory_space(), comm_particles );
+            halo_buffer_aosoa_gather.stop( fraction );
+        }
+
+        // Run preallocated slice buffer gather tests and time the ensemble.
+        comm_particles = Cabana::create_mirror_view_and_copy(
+            comm_memory_space(), particles );
+        auto s0 = Cabana::slice<0>( comm_particles );
+        auto s1 = Cabana::slice<1>( comm_particles );
+        auto s2 = Cabana::slice<2>( comm_particles );
+        auto s3 = Cabana::slice<3>( comm_particles );
+        auto gather_s0 = createGather( halo, s0, overallocation );
+        auto gather_s1 = createGather( halo, s1, overallocation );
+        auto gather_s2 = createGather( halo, s2, overallocation );
+        auto gather_s3 = createGather( halo, s3, overallocation );
+        for ( int t = 0; t < num_run; ++t )
+        {
+            halo_buffer_slice_gather.start( fraction );
+            comm_particles = Cabana::create_mirror_view_and_copy(
+                comm_memory_space(), particles );
+
+            auto s0 = Cabana::slice<0>( comm_particles );
```

Comment:
Don't these `s0`, `s1`, etc. mirror the definitions above? If we have the gather/scatter objects store the aosoa/slice they are working on then these wouldn't be needed anyway.

Reply:
Fixed to match above